test/unit/iteration_test.rb

```diff
@@ -251,6 +251,75 @@ def test_global_max_job_runtime
     end
   end
 
+  def test_per_class_max_job_runtime_with_no_global_set
+    freeze_time
+    with_global_max_job_runtime(nil) do
```

Comment:
I find this a bit confusing, because I'd expect "with no global set" to mean that we don't touch `JobIteration.max_job_runtime` at all, not that we explicitly set it to `nil`. If we were to change what not setting the default means at a future point in time (e.g. if `JobIteration.max_job_runtime` defaults to something other than `nil` at a future point), this would fail -- although arguably it would be easy enough to reason about why 😅 

Is there a reason we need to use `with_global_max_job_runtime` here -- can we just rely on what the configured default is, without modifying `JobIteration.max_job_runtime`?